test/Concurrency/sendable_conformance_checking.swift

```diff
@@ -204,3 +204,16 @@ public struct TestSendableWitnesses2 : NoSendableReqs {
   var prop = globalFn // Ok (no warnings)
   static let staticProp = globalFn // Ok (no warnings)
 }
+
+// @preconcurrency attributes to make it akin to an imported Obj-C API
+@preconcurrency @MainActor public protocol EscapingSendableProtocol {
+  // expected-complete-and-tns-note @+1 {{protocol requires function 'f(handler:)' with type '(@escaping @MainActor @Sendable (Int) -> Void) -> ()'}}
+  @preconcurrency func f(handler: @escaping @MainActor @Sendable (Int) -> Void)
+}
+
+// expected-complete-and-tns-error @+2 {{type 'TestEscapingOnly' does not conform to protocol 'EscapingSendableProtocol'}}
```

Comment:
I think this should be a warning and not an error due to the `@preconcurrency` on the protocol requirement. The effect of `@preconcurrency` on a declaration is that it downgrades concurrency errors to warnings.

I think you should add a test that has this protocol in a separate module, that's then imported into the file with the conforming type via a `@preconcurrency` import, which should suppress this error entirely.

Reply:
@hborla I did not change anything about that error in this PR, so it was an error before this, but are you suggesting that you want me in this PR to change it from warning to error?

I did try reconstructing the test with the protocol in a separate module, but under strict concurrency and with `@preconcurrency import`, these errors are still produced.